src/libraries/System.Runtime/tests/System.Reflection.Tests/PropertyInfoTests.cs

```diff
@@ -342,6 +342,14 @@ public void PropertyType(Type type, string name, Type expected)
             Assert.Equal(expected, propertyInfo.PropertyType);
         }
 
+        [Theory]
+        [InlineData(typeof(BaseClass), nameof(BaseClass.BoolProperty))]
+        [InlineData(typeof(SubClass), nameof(SubClass.BoolProperty))]
+        public void HiddenPublicProperty(Type type, string name)
+        {
+            Assert.NotNull(type.GetProperty(name));
```

Comment:
Also. this is testing GerProperty with `BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public`. Is it worth testing `GetProperty` with e.g. `BindingFlags.Public | BindingFlags.NonPublic`?

Reply:
Same for `GetProperties`  - test `BindingFlags.Public`, `BindingFlags.Public | BindingFlags.NonPublic`, `BindingFlags.NonPublic`?